core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -1310,24 +1310,26 @@ public static PlanAndMappings coerce(PlanBuilder subPlan, List<Expression> expre
         Assignments.Builder assignments = Assignments.builder();
         assignments.putIdentities(subPlan.getRoot().getOutputSymbols());
 
-        ImmutableMap.Builder<NodeRef<Expression>, Symbol> mappings = ImmutableMap.builder();
+        Map<NodeRef<Expression>, Symbol> mappings = new HashMap<>();
         for (Expression expression : expressions) {
             Type coercion = analysis.getCoercion(expression);
 
-            if (coercion != null) {
-                Type type = analysis.getType(expression);
-                Symbol symbol = symbolAllocator.newSymbol(expression, coercion);
+            if (!mappings.containsKey(NodeRef.of(expression))) {
```

Comment:
Why would `expressions` contain duplicates (in identity sense)?

is it because `SELECT exp ... GROUP BY 1, 1`  gets "converted" into `... GROUP BY exp, exp`?

add explanation to the code.

Reply:
Yes. It's because the `1` are treated as "references" to an expression. From the point of view of this method, there's no requirement that the expressions be unique in the provided list, so it needs to protect against duplicates singe it's using a structure that doesn't allow them.